### PR TITLE
UTXO cache trace points and sample scripts

### PR DESCRIPTION
**Pull request by Naviabheeman**
This PR ports utxo cache trace points from bitcoin core to Tapyrus-core. It includes the python functional tests for the tracepoints using bcc(python-bpfcc). However these scripts do not run successfully in the ubuntu 18 used in CI or in the docker environment. SO these are disabled now. the bpftrce scripts(.bt) work fine.

### CODE REVIEW

File: test/functional/interface_usdt_utxocache.py [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+        self.log.info(
+            f"check that we successfully traced {EXPECTED_HANDLE_ADD_SUCCESS} adds and {EXPECTED_HANDLE_SPENT_SUCCESS} spent")
+        assert_equal(0, len(expected_utxocache_adds))
+        assert_equal(0, len(expected_utxocache_spents))
+        assert_equal(EXPECTED_HANDLE_ADD_SUCCESS, handle_add_succeeds)
+        assert_equal(EXPECTED_HANDLE_SPENT_SUCCESS, handle_spent_succeeds)
+
+    def test_flush(self):
+        """ Tests the utxocache:utxocache_flush tracepoint API.
+            See https://github.com/bitcoin/bitcoin/blob/master/doc/tracing.md#tracepoint-utxocacheflush"""
```
Comment (azuchi) [inline test/functional/interface_usdt_utxocache.py:339]: This needs to be replaced with a valid link on the Tapyrus Core repository.

File: doc/build-unix.md [hunk 2]
```diff
@@ -117,11 +117,25 @@ Build requirements:
 
     sudo dnf install gcc-c++ libtool make autoconf automake python3
 
+<<<<<<< HEAD
+=======
+<<<<<<< HEAD
+Now, you can either build from self-compiled [depends](/depends/README.md) or install the required dependencies:
+=======
```
Comment (azuchi) [inline doc/build-unix.md:124]: Looks like a merge mistake.

File: doc/build-unix.md [hunk 3]
```diff
@@ -117,11 +117,25 @@ Build requirements:
...
+>>>>>>> 82a6d0d55a (add tracepoints to utco cache)
 User-Space, Statically Defined Tracing (USDT) dependencies:
 
     sudo dnf install systemtap
 
 Optional:
+<<<<<<< HEAD
+=======
+>>>>>>> 79687c640f (add tracepoints to utco cache)
+>>>>>>> 82a6d0d55a (add tracepoints to utco cache)
```
Comment (azuchi) [inline doc/build-unix.md:138]: Looks like a merge mistake

File: contrib/tracing/README.md [hunk 4]
```diff
@@ -163,3 +163,109 @@ Warning: incomplete message (only 32568 out of 53552 bytes)! inbound msg 'tx' fr
...
+
+Histogram of block connection times in milliseconds (ms).
+@durations: 
+[0]                  397 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@|
+
+
+
+```
+
+### log_utxocache_flush.py
```
Comment (azuchi) [inline contrib/tracing/README.md:229]: This file not found.

File: contrib/tracing/README.md [hunk 5]
```diff
@@ -163,3 +163,109 @@ Warning: incomplete message (only 32568 out of 53552 bytes)! inbound msg 'tx' fr
...
+
+```
+Logging utxocache flushes. Ctrl-C to end...
+Duration (µs)   Mode       Coins Count     Memory Usage    Prune
+730451          IF_NEEDED  22990           3323.54 kB      True
+637657          ALWAYS     122320          17124.80 kB     False
+81349           ALWAYS     0               1383.49 kB      False
+```
+
+### log_utxos.bt
```
Comment (azuchi) [inline contrib/tracing/README.md:246]: This file not found.

**Review by azuchi — CHANGES_REQUESTED** [review]

File: contrib/tracing/README.md (hunk 4 above)
Comment (Naviabheeman) [reply contrib/tracing/README.md:229]: fixed!

File: contrib/tracing/README.md (hunk 5 above)
Comment (Naviabheeman) [reply contrib/tracing/README.md:246]: fixed!

**Review by azuchi — APPROVED** [review]

**Pull request merged**